lib/spack/spack/schema/env.py

```diff
@@ -133,3 +136,22 @@
         }
     }
 }
+
+
+def update(data):
```

Comment:
`data` is a `SingleFileScope`. IMO this should move to the `config` module.

Reply:
(marking as resolved) As I think about it again, I feel like my complaint is esoteric: the environment schema is largely a merged schema; if other merged schemas (i.e. incorporating packages.yaml, compilers.yaml, etc) were created in the future, the current implementation might lead to some code duplication. For now this is not a problem. If merged schemas appear in the future, I think it will be clear where the duplication arises and this code can move then.